src/libpmemfile/preload.c

```diff
@@ -2794,7 +2886,12 @@ pmemfile_preload_constructor(void)
 	env_str = getenv("PMEMFILE_PRELOAD_PROCESS_SWITCHING");
 	process_switching = env_str ? env_str[0] == '1' : 0;
 
-	establish_mount_points(getenv("PMEMFILE_POOLS"));
+	assert(pool_count == 0);
+	struct stat kernel_cwd_stat;
+	stat_cwd(&kernel_cwd_stat);
+
+	detect_mount_points(&kernel_cwd_stat);
+	establish_mount_points(getenv("PMEMFILE_POOLS"), &kernel_cwd_stat);
```

Comment:
Well, the naming is weird here.
Both of these functions detect mount points (from different sources), and both establish them.

Reply:
Detect_mount_points gets the information about mount points from system.
Establish_mount_points doesn't detect anything. getenv("PMEMFILE_POOLS") is the detection part.

So I don't see any problem with naming here. I'm open to suggestions though.